frontend/packages/pipelines-plugin/src/components/pipelineruns/hooks/useTektonResults.ts

```diff
@@ -0,0 +1,199 @@
+                  ? (() => {
+                      // ensure we can only call this once
+                      let executed = false;
+                      return () => {
+                        if (!disposed && !executed) {
+                          executed = true;
+                          // trigger the update
+                          setNextPageToken(token);
+                          return true;
+                        }
+                        return false;
+                      };
+                    })()
+                  : undefined,
+              ]);
+            }
+          }
+        } catch (e) {
+          if (!disposed) {
+            if (nextPageToken) {
+              setResult((cur) => [cur[0], cur[1], e, undefined]);
+            } else {
+              setResult([[], false, e, undefined]);
+            }
+          }
+        }
+      })();
+      return () => {
+        disposed = true;
+      };
+    }
+  }, [namespace, options, nextPageToken, localCacheKey, getRuns]);
+  return result;
+};
+
+export const useTRPipelineRuns = (
+  namespace: string,
+  options?: TektonResultsOptions,
+  cacheKey?: string,
+): [PipelineRunKind[], boolean, unknown, GetNextPage] =>
+  useTRRuns<PipelineRunKind>(getPipelineRuns, namespace, options, cacheKey);
+
+export const useTRTaskRuns = (
+  namespace: string,
+  options?: TektonResultsOptions,
+  cacheKey?: string,
+): [TaskRunKind[], boolean, unknown, GetNextPage] =>
+  useTRRuns<TaskRunKind>(getTaskRuns, namespace, options, cacheKey);
+
+export const useGetPipelineRuns = (ns: string, options?: { name: string; kind: string }) => {
+  let selector: Selector;
+
+  if (options?.kind === 'Pipeline') {
+    selector = { matchLabels: { 'tekton.dev/pipeline': options?.name } };
+  }
+  if (options?.kind === 'Repository') {
+    selector = { matchLabels: { [RepositoryLabels[RepositoryFields.REPOSITORY]]: options?.name } };
+  }
+  const [resultPlrs, resultPlrsLoaded, resultPlrsLoadError, getNextPage] = useTRPipelineRuns(
+    ns,
+    options && {
+      selector,
+    },
+  );
+  const [k8sPlrs, k8sPlrsLoaded, k8sPlrsLoadError] = useK8sWatchResource<PipelineRunKind[]>({
+    isList: true,
+    kind: referenceForModel(PipelineRunModel),
+    namespace: ns,
+    ...(options ? { selector } : {}),
+  });
+  const mergedPlrs =
+    (resultPlrsLoaded || k8sPlrsLoaded) && !k8sPlrsLoadError
+      ? uniqBy([...k8sPlrs, ...resultPlrs], (r) => r.metadata.name)
+      : [];
+  return [
+    mergedPlrs,
+    resultPlrsLoaded || k8sPlrsLoaded,
+    resultPlrsLoadError || k8sPlrsLoadError,
+    getNextPage,
+  ];
+};
+
+export const useGetTaskRuns = (
+  ns: string,
+  pipelineRunName?: string,
+): [TaskRunKind[], boolean, unknown, GetNextPage] => {
+  const [k8sTaskRuns, k8sTaskRunsLoaded, k8sTaskRunsLoadError] = useTaskRuns(ns, pipelineRunName);
+  const [
+    resultTaskRuns,
+    resultTaskRunsLoaded,
+    resultTaskRunsLoadError,
+    getNextPage,
+  ] = useTRTaskRuns(ns);
```

Comment:
Here you are passing only the namespace but not options to select for particular pipelineRun.